brewing recipe tag bug

### Issue description

There seemed to be a bug wen I tried to put a tag into a brewing recipe.


> Task :compileJava FAILED 
C:\Users\grumm\Desktop\just-another-vanilla-addon-master\just-another-vanilla-addon-master\src\main\java\cn\mcmod\java\recipes\brewing\WenTheCaementBrewingRecipe.java:36: ����: �Ҳ������� 
      return ItemTags.getCollection().getTagByID(new ResourceLocation("forge:webs")).contains(ingredient.getItem()); 
                     ^ 
  ����:   ���� getCollection() 
  λ��: �� ItemTags 
1 ������ 
FAILURE: Build failed with an exception.

### How to reproduce this issue?

Put a tag into the first slot of a brewing recipe.
I don't know what will happen if I put it into another slot, but I think it'll be similar.

### Operating system

Windows

### Details

MCreatorEAP20221b12711
Forge-1.18.2

### Example workspace

[_No response_](https://gitee.com/grumm9472/just-another-vanilla-addon/tree/1.18.2/)

### Logs

Executing Gradle task: build 
Build info: MCreator 2022.1.12711 EAP (12711), forge-1.18.2, 64-bit, 16298 MB, Windows 10, JVM 17.0.1, JAVA_HOME: C:\Users\grumm\Downloads\Compressed\MCreatorEAP20221b12711\jdk, started on: 2022-03-15-10:38:43 
  
> Configure project : 
The code of this workspace uses official obfuscation mappings provided by Mojang. These mappings fall under their associated license you should be fully aware of. 
(c) 2020 Microsoft Corporation. These mappings are provided "as-is" and you bear the risk of using them. You may copy and use the mappings for development purposes, 
but you may not redistribute the mappings complete and unmodified. Microsoft makes no warranties, express or implied, with respect to the mappings provided here. 
Use and modification of this document or the source code (in any form) of Minecraft: Java Edition is governed by the Minecraft End User License Agreement available 
at https://account.mojang.com/documents/minecraft_eula. 
  
> Task :compileJava FAILED 
C:\Users\grumm\Desktop\just-another-vanilla-addon-master\just-another-vanilla-addon-master\src\main\java\cn\mcmod\java\recipes\brewing\WenTheCaementBrewingRecipe.java:36: ����: �Ҳ������� 
      return ItemTags.getCollection().getTagByID(new ResourceLocation("forge:webs")).contains(ingredient.getItem()); 
                     ^ 
  ����:   ���� getCollection() 
  λ��: �� ItemTags 
1 ������ 
FAILURE: Build failed with an exception. 
* What went wrong: 
Execution failed for task ':compileJava'. 
> Compilation failed; see the compiler error output for details. 
* Try: 
> Run with --stacktrace option to get the stack trace. 
> Run with --info or --debug option to get more log output. 
> Run with --scan to get full insights. 
* Get more help at https://help.gradle.org 
BUILD FAILED in 5s 
1 actionable task: 1 executed 
  
BUILD FAILED 
Task completed in 5 seconds

### Issue tracker rule checks (please read carefully)

- [X] I have checked that my problem <a href='https://github.com/MCreator/MCreator/issues?q=is%3Aissue' target='_blank'>is not already reported</a>
- [X] I have checked that my problem is not covered on <a href='https://mcreator.net/support/knowledgebase' target='_blank'>Knowledge Base</a> or on <a href='https://mcreator.net/wiki' target='_blank'>MCreator's Wiki</a>
- [X] I have checked that my written content does not violate the <a href='https://mcreator.net/wiki/general-publishing-guidelines' target='_blank'>publishing guidelines</a>

--- a/plugins/generator-1.18.2/forge-1.18.2/templates/brewingrecipe.java.ftl
+++ b/plugins/generator-1.18.2/forge-1.18.2/templates/brewingrecipe.java.ftl
@@ -46,7 +46,7 @@ package ${package}.recipes.brewing;
 		return (inputItem == Items.POTION || inputItem == Items.SPLASH_POTION || inputItem == Items.LINGERING_POTION)
 			&& PotionUtils.getPotion(input) == ${generator.map(data.brewingInputStack?replace("POTION:",""), "potions")};
 		<#elseif data.brewingInputStack?starts_with("TAG:")>
-		return ItemTags.getCollection().getTagByID(new ResourceLocation("${data.brewingInputStack?replace("TAG:","")}")).contains(input.getItem());
+		return ItemTags.create(new ResourceLocation("${data.brewingInputStack?replace("TAG:","")}")).test(input.getItem());
 		<#else>
 		return input.getItem() == ${mappedMCItemToItem(data.brewingInputStack)};
 		</#if>
@@ -55,7 +55,7 @@ package ${package}.recipes.brewing;
 	@Override
 	public boolean isIngredient(ItemStack ingredient) {
 		<#if data.brewingIngredientStack?starts_with("TAG:")>
-		return ItemTags.getCollection().getTagByID(new ResourceLocation("${data.brewingIngredientStack?replace("TAG:","")}")).contains(ingredient.getItem());
+		return ItemTags.create(new ResourceLocation("${data.brewingIngredientStack?replace("TAG:","")}")).test(ingredient.getItem());
 		<#else>
 		return ingredient.getItem() == ${mappedMCItemToItem(data.brewingIngredientStack)};
 		</#if>

--- a/plugins/generator-1.18.2/forge-1.18.2/templates/brewingrecipe.java.ftl
+++ b/plugins/generator-1.18.2/forge-1.18.2/templates/brewingrecipe.java.ftl
@@ -46,7 +46,7 @@ package ${package}.recipes.brewing;
 		return (inputItem == Items.POTION || inputItem == Items.SPLASH_POTION || inputItem == Items.LINGERING_POTION)
 			&& PotionUtils.getPotion(input) == ${generator.map(data.brewingInputStack?replace("POTION:",""), "potions")};
 		<#elseif data.brewingInputStack?starts_with("TAG:")>
-		return ItemTags.create(new ResourceLocation("${data.brewingInputStack?replace("TAG:","")}")).test(input.getItem());
+		return Ingredient.of(ItemTags.create(new ResourceLocation("${data.brewingInputStack?replace("TAG:","")}"))).test(input);
 		<#else>
 		return input.getItem() == ${mappedMCItemToItem(data.brewingInputStack)};
 		</#if>
@@ -55,7 +55,7 @@ package ${package}.recipes.brewing;
 	@Override
 	public boolean isIngredient(ItemStack ingredient) {
 		<#if data.brewingIngredientStack?starts_with("TAG:")>
-		return ItemTags.create(new ResourceLocation("${data.brewingIngredientStack?replace("TAG:","")}")).test(ingredient.getItem());
+		return Ingredient.of(ItemTags.create(new ResourceLocation("${data.brewingIngredientStack?replace("TAG:","")}"))).test(ingredient);
 		<#else>
 		return ingredient.getItem() == ${mappedMCItemToItem(data.brewingIngredientStack)};
 		</#if>

